src/core/hle/D3D8/XbPixelShader.cpp

```diff
@@ -1395,7 +1411,17 @@ void DxbxUpdateActivePixelShader() // NOPATCH
 		  frontfaceFactor = (pg->KelvinPrimitive.SetFrontFace == 0x900) ? 1.0f : -1.0f;
 	  }
 	  fColor[PSH_XBOX_CONSTANT_FRONTFACE_FACTOR].r = frontfaceFactor;
-
+	  fogEnable = (NV2ARenderStates.GetXboxRenderState(xbox::_X_D3DRENDERSTATETYPE::X_D3DRS_FOGENABLE) > 0 == Fog_Enabled) ? Fog_Enabled : Fog_Disabled;
```

Comment:
Please remove the ` == Fog_Enabled` part, because its inadvertently inverting the result while the "comparison above zero" already results in a boolean. (Turning that boolean into one of the "enabled" or "disabled" constants in the rest of the ternary expression is fine, besides their name and declaration which I commented on elsewhere.)

Reply:
@PatrickvL the GetXboxRenderState() returns a DWORD instead of float. original code should work correctly. 
fogEnable = NV2ARenderStates.GetXboxRenderState(xbox::_X_D3DRENDERSTATETYPE::X_D3DRS_FOGENABLE) !=false;
or even directly assign the renderstate to fogEnable should work.

Reply:
Ah, that I overlooked, probably because all assignments are targeting float variants, so apparently those are all doing an implicit (instead of an explicit) type-cast.

Still, my remark would make it somewhat more clear that this specific variable contains a boolean. What do you think is wisest to do here? Be more explicit or rely on implicit behaviour?

(Also, remember that later on when the port to Direct3D 11 takes shape, shader variables can become integral or boolean besides the current float-only type.)

Reply:
how about this 
const FloatTrue =1.0
const FloatFalse = 0;
fogEnable = NV2ARenderStates.GetXboxRenderState(xbox::_X_D3DRENDERSTATETYPE::X_D3DRS_FOGENABLE) !=false?FloatTrue:FloatFalse;

Reply:
That can work, sure.
Those two consts would be good to have indeed.

Except `X != false` is a double negative
While `X > 0` is more easy on my eyes.

So medievil, you choose!

Reply:
x > 0 it is...
both are readable and understandable, chose it because it is less characters/space....lol

I placed the FloatTrue/False inside Direct3d9.h so it can be accessed by all of the graphics code